db/migrate/20170602111120_add_timestamps_to_documents.rb

```diff
@@ -0,0 +1,117 @@
+class AddTimestampsToDocuments < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def change
+    add_timestamps(:documents, null: true) unless column_exists?(:documents, :created_at)
+    puts "> Timestamp fields added."
+    puts "> Updating multiple-locale documents without editions..."
+
+    update_multiple_locale_documents_without_editions
+    puts "> Multiple-locale documents without editions updated."
+    puts "> Updating remaining documents..."
+
+    remaining_documents = Document.where(created_at: nil).count
+
+    Document.where(created_at: nil).find_each.with_index do |document, index|
+      editions = Edition.where(document_id: document.id).order(:user_facing_version)
+      if editions.exists?
+        update_timestamps_from_editions(document, editions)
+      else
+        update_timestamps_from_events(document)
+      end
+      puts "> #{index += 1}/#{remaining_documents} documents updated."
+    end
+
+    puts "> All remaining documents updated."
+    puts "> Setting NOT NULL constraint on created_at and updated_at..."
+    change_column_null :documents, :created_at, false
+    change_column_null :documents, :updated_at, false
+  end
```

Comment:
Do you think the timestamps should be set to nil, or just remove the columns entirely?